contentprovider.go

```diff
@@ -218,8 +217,16 @@ func (p *contentProvider) fillContentMatches(ms []*candidateMatch) []LineMatch {
 				Offset:      m.byteOffset,
 				LineOffset:  int(m.byteOffset) - lineStart,
 				MatchLength: int(m.byteMatchSz),
-				SymbolInfo:  m.symbolInfo,
 			}
+			if m.symbol {
+				start := p.id.fileEndSymbol[p.idx]
+				fragment.SymbolInfo = p.id.symbols.data(start + m.symbolIdx)
+				if fragment.SymbolInfo != nil {
+					sec := p.docSections()[m.symbolIdx]
+					fragment.SymbolInfo.Sym = string(data[sec.Start:sec.End])
```

Comment:
maybe for follow-up PR: Should we make SymbolInfo fields all `[]byte`? I see above for `.Line` we use a []byte which avoids the need to copy the data into a string.